include/core/thermal_conductivity_model.h

```diff
@@ -188,4 +188,122 @@ class ThermalConductivityLinear : public ThermalConductivityModel
   const double B;
 };
 
+/**
+ * @brief ThermalConductivityPhaseChange Implements a phase-dependant thermal conductivity
+ */
+class ThermalConductivityPhaseChange : public ThermalConductivityModel
+{
+public:
+  /**
+   * @brief Default constructor
+   */
+  ThermalConductivityPhaseChange(const double thermal_conductivity_s,
+                                 const double thermal_conductivity_l,
+                                 const double T_solidus,
+                                 const double T_liquidus)
+    : thermal_conductivity_s(thermal_conductivity_s)
+    , thermal_conductivity_l(thermal_conductivity_l)
+    , T_solidus(T_solidus)
+    , T_liquidus(T_liquidus)
+  {}
+
+  /**
+   * @brief value Calculates the value the thermal conductivity
+   * @param fields_value Value of the various field on which the thermal conductivity depends.
+   * @return value of the thermal conductivity calculated with the fields_value.
+   */
+  double
+  value(const std::map<field, double> &fields_value) override
+  {
+    double thermal_conductivity;
+    // Thermal conductivity of solid phase
+    if (fields_value.at(field::temperature) < T_solidus)
+      thermal_conductivity = thermal_conductivity_s;
+    // Thermal conductivity of liquid phase
+    else if (fields_value.at(field::temperature) > T_liquidus)
+      thermal_conductivity = thermal_conductivity_l;
+    // Mean value of the conductivities of the solid and liquid phases
+    else
+      {
+        const double l_frac =
```

Comment:
Can you add the calculation of the liquid fraction in a separate function that is inline instead of calculating here. This way you can copy the function from the other phase change models.

Reply:
done.